src/highdicom/seg/sop.py

```diff
@@ -1823,38 +1920,118 @@ def segmented_property_types(self) -> List[CodedConcept]:
 
         return types
 
-    def _get_src_uid_index(self, sop_instance_uid: str) -> int:
-        ind = np.argwhere(self._source_sop_instance_uids == sop_instance_uid)
-        if len(ind) == 0:
-            raise KeyError(
-                f'No such source frame: {sop_instance_uid}'
-            )
-        return ind.item()
+    def _check_sop_uid_exists(self, sop_instance_uid: str) -> None:
+        """Checks whether a SOP Instance UID is in the database.
+
+        Parameters
+        ----------
+        sop_instance_uid: str
+            SOP Instance UID value to check.
+
+        Raises
+        ------
+        KeyError:
+            If the input sop_instance_uid is not in the database.
+
+        """
+        cur = self._db_con.cursor()
+        res = cur.execute(
+            "SELECT SOPInstanceUID FROM InstanceUIDs "
+            f"WHERE SOPInstanceUID='{sop_instance_uid}' LIMIT 1"
+        )
+        first = res.fetchone()
+        if first is None:
+            raise KeyError("No such source frame: {sop_instance_uid}.")
+
+    def _create_temporary_segment_table(
+        self,
+        segment_numbers: Sequence[int],
+        combine_segments: bool,
+        relabel: bool,
+    ) -> None:
+        """Create a temporary table of desired segment in the SQLite DB.
+
+        Assumes the self._db_con object is currently being used as a context
```

Comment:
That seems to be quite an assumption and asking for troubling..

Reply:
This no longer applies after all the of the recent refactoring. All methods on the new `SegDBManager` class handle their own commits to the database